src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -1100,6 +1101,41 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     // not as a fragment. Nested arrays on the other hand will be treated as
     // fragment nodes. Recursion happens at the normal flow.
 
+    if (ReactFeatureFlags.disableNewFiberFeatures) {
+      // Support only the subset of return types that Stack supports. Treat
+      // everything else as empty, but log a warning.
+      if (typeof newChild === 'object' && newChild !== null) {
+        switch (newChild.$$typeof) {
+          case REACT_ELEMENT_TYPE:
+            return placeSingleChild(reconcileSingleElement(
+              returnFiber,
+              currentFirstChild,
+              newChild,
+              priority
+            ));
+
+          case REACT_PORTAL_TYPE:
+            return placeSingleChild(reconcileSinglePortal(
+              returnFiber,
+              currentFirstChild,
+              newChild,
+              priority
+            ));
+        }
+      }
+
+      if (__DEV__) {
+        const Component = returnFiber.type;
+        warning(
```

Comment:
I'll cherry-pick everything except the last commit, in favor of the way I did it here, for the reason Sebastian gave here: https://github.com/facebook/react/pull/8648#discussion_r94506177